### PR TITLE
tests: check for `asdf` and `npm` with goss

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3763

this change the way we perform GOSS check by impersonating the jenkins user

### CODE REVIEW

**Comment by smerle33** [comment]
updatecli error due to chicken and egg : `ERROR: something went wrong in target "updateASDFVersionInGoss" : "couldn't find key \"$.command.asdf.stdout[0]\" from file \"goss/goss.yaml\""`
local test without `scmid: default` looks good : 
```
 Bump ASDF version:
        Source:
                ✔ [lastReleaseVersion] Get the latest ASDF version
        Target:
                ⚠ [updateASDFVersionInGoss] Update the ASDF version in the goss test
 ```

File: build-jenkins-agent-ubuntu.pkr.hcl [hunk 1]
```diff
@@ -46,10 +46,12 @@ build {
   }
 
   provisioner "shell" {
+    execute_command  = "{{ .Vars }} sudo -E su - jenkins -c \"bash -eu '{{ .Path }}'\""
     inline = [
-      "set -xeu",
+      "source /home/jenkins/.asdf/asdf.sh",
+      "asdf version",
```
Comment (dduportal) [inline build-jenkins-agent-ubuntu.pkr.hcl:52]:
```suggestion
```

(sounds like a former debug step)

File: build-jenkins-agent-ubuntu.pkr.hcl [hunk 2]
```diff
@@ -46,10 +46,12 @@ build {
   }
 
   provisioner "shell" {
+    execute_command  = "{{ .Vars }} sudo -E su - jenkins -c \"bash -eu '{{ .Path }}'\""
     inline = [
-      "set -xeu",
+      "source /home/jenkins/.asdf/asdf.sh",
```
Comment (dduportal) [inline build-jenkins-agent-ubuntu.pkr.hcl:51]:
```suggestion
      "source /home/jenkins/.asdf/asdf.sh", # Required as this is a non-interactive and non-login `bash`
```

**Review by dduportal — CHANGES_REQUESTED** [review]
Good job 👍 

A few debug leftovers and a nitpick suggestion for a comment (only leftovers are required to be updated)

**Review by dduportal — APPROVED** [review]

**Pull request merged**